api/v1alpha3/cloudstackcluster_webhook.go

```diff
@@ -54,19 +55,20 @@ var _ webhook.Validator = &CloudStackCluster{}
 func (r *CloudStackCluster) ValidateCreate() error {
 	cloudstackclusterlog.Info("validate create", "name", r.Name)
 
-	var (
-		errorList field.ErrorList
-		spec      = r.Spec
-	)
+	var errorList field.ErrorList
 
 	// IdentityRefs must be Secrets.
-	if spec.IdentityRef != nil && spec.IdentityRef.Kind != defaultIdentityRefKind {
+	if r.Spec.IdentityRef != nil && r.Spec.IdentityRef.Kind != defaultIdentityRefKind {
 		errorList = append(errorList, field.Forbidden(field.NewPath("spec", "identityRef", "kind"), "must be a Secret"))
 	}
 
+	if (r.Spec.Account == "") != (r.Spec.Domain == "") {
```

Comment:
Require both account and domain if set.

Reply:
isn't domain itself sufficient for performing all the necessary operations? Why do we also need account? Also is this an account ID? Or the account name?

Reply:
Sorry, I worded it wrong.

If account is set, domain must be set too.

https://cloudstack.apache.org/api/apidocs-4.16/apis/deployVirtualMachine.html

It's a restriction imposed by the cloudstack api itself.

> 

account | an optional account for the virtual machine. Must be used with domainId. | false
-- | -- | --

Reply:
right so wouldn't this check fail if only domain is provided, even though that's an acceptable situation?